test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/PageActionInvokerProviderTest.cs

```diff
@@ -299,6 +362,238 @@ public void OnProvidersExecuting_UpdatesEntriesWhenActionDescriptorProviderColle
+        }
+        
+        [Theory]
+        [InlineData(typeof(TestSetPageWithModel))]
+        [InlineData(typeof(TestSetPageModel))]
+        public void PopulateHandlerMethodDescriptors_FindsAll(Type pageType)
+        {
+            // Arrange
+            var descriptor = new PageActionDescriptor()
+            {
+                RelativePath = "Path1",
+                FilterDescriptors = new FilterDescriptor[0],
+                ViewEnginePath = "/Views/Deeper/Index.cshtml"
+            };
+
+            var actionDescriptor = CreateCompiledPageActionDescriptor(descriptor, pageType);
+
+            var type = actionDescriptor.ModelTypeInfo ?? actionDescriptor.PageTypeInfo;
+
+            // Act
+            PageActionInvokerProvider.PopulateHandlerMethodDescriptors(type, actionDescriptor);
+
+            // Assert
+            Assert.Equal(2, actionDescriptor.HandlerMethods.Count);
+        }
+
+        [Fact]
+        public void GetPageStartFactories_FindsFullHeirarchy()
+        {
+            // Arrange
+            var descriptor = new PageActionDescriptor()
+            {
+                RelativePath = "Path1",
+                FilterDescriptors = new FilterDescriptor[0],
+                ViewEnginePath = "/Views/Deeper/Index.cshtml"
+            };
+            var loader = new Mock<IPageLoader>();
+            loader.Setup(l => l.Load(It.IsAny<PageActionDescriptor>()))
+                .Returns(CreateCompiledPageActionDescriptor(descriptor, typeof(TestPageModel)));
+            var descriptorCollection = new ActionDescriptorCollection(new[] { descriptor }, version: 1);
+            var actionDescriptorProvider = new Mock<IActionDescriptorCollectionProvider>();
+            actionDescriptorProvider.Setup(p => p.ActionDescriptors).Returns(descriptorCollection);
+
+            var fileProvider = new TestFileProvider();
+            fileProvider.AddFile("/View/Deeper/Not_PageStart.cshtml", "page content");
+            fileProvider.AddFile("/View/Wrong/_PageStart.cshtml", "page content");
+            fileProvider.AddFile("/_PageStart.cshtml", "page content ");
+            fileProvider.AddFile("/Views/_PageStart.cshtml", "@page starts!");
+            fileProvider.AddFile("/Views/Deeper/_PageStart.cshtml", "page content");
+
+            var razorProject = CreateRazorProject(fileProvider);
+
+            var invokerProvider = CreateInvokerProvider(
+                loader.Object,
+                actionDescriptorProvider.Object,
+                pageProvider: null,
+                modelProvider: null,
+                razorPageFactoryProvider: CreateRazorPageFactoryProvider(),
+                razorProject: razorProject);
+
+            var compiledDescriptor = CreateCompiledPageActionDescriptor(descriptor);
+
+            // Act
+            var factories = invokerProvider.GetPageStartFactories(compiledDescriptor);
+
+            // Assert
+            Assert.Equal(3, factories.Count);
```

Comment:
You want to verify that the PageFactoryProvider is invoked with the right values of `_PageStart`.